### ISSUE TITLE
Panning with mouse using Axes3d in plt.show() is laggy

### ISSUE THREAD

**Issue by joemasiero** [matplotlib/matplotlib#9301]
### Bug report

**Bug summary**

When plotting a 3D dataset using Axes3D and plt.show(), the plot does not respond correctly to the pan controls from the mouse.  The when pressed and held, mouse position is recorded, but not re-drawn until the button is released and pressed again.  This only applies to panning.  Zoom does respond and redraw in real time as the mouse is moved.

**Code for reproduction**

```python
# Paste your code here
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D

fig=plt.figure(figsize=[8,8])
ax=fig.gca(projection='3d')
ax.view_init(20,0)
ax.plot([1,2,3,4,5,6,7],[3,5,3,5,3,5,3],[5,3,1,3,5,3,1],marker='o',color='black')
plt.draw()
plt.show()

```

**Actual outcome**

using button 3, plot responds in real-time to zoom requests and zoom follows the mouse.  However, using button 1, the mouse position is recorded, but the plot does not update until the button is released and then pressed again.

**Expected outcome**

Pan and zoom with mouse worked as expected in matplotlib 1.4, under older version of python (sorry, don't have the number off hand)

**Matplotlib version**
  * Operating system: MAC OSX 10.12.6 
  * Matplotlib version: 2.0.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): MacOSX
  * Python version: 2.7.14

local install of python from pkg, matplotlib installed from pip

**Comment by dstansby** [comment]
Pretty sure this is the same as #8814

**Comment by joemasiero** [comment]
Thanks. Yes, it appears to be the same.  Sorry about missing that before posting.  The fix you suggested in the #8814 thread seems to work for me.
